core/src/test/java/de/unistuttgart/iaas/amyassist/amy/core/configuration/EnvironmentConfigurationLoaderTest.java

```diff
@@ -0,0 +1,319 @@
+		this.envVars.put("config_1_key2", "value12");
+		this.envVars.put("config_2_key1", "value21");
+
+		Properties orig1 = new Properties();
+		Properties orig2 = new Properties();
+
+		orig1.setProperty("key1", "orig11");
+		orig1.setProperty("key2", "orig12");
+		orig2.setProperty("key1", "orig21");
+
+		Properties res1 = this.loader.load("config.1", orig1);
+		Properties res2 = this.loader.load("config.2", orig2);
+
+		Assertions.assertEquals("value11", res1.getProperty("key1"));
+		Assertions.assertEquals("value12", res1.getProperty("key2"));
+		Assertions.assertEquals("value21", res2.getProperty("key1"));
+	}
+
+	/**
+	 * Tests the ignore case
+	 * <p>
+	 * Test method for
+	 * {@link de.unistuttgart.iaas.amyassist.amy.core.configuration.EnvironmentConfigurationLoaderImpl#load(java.lang.String, java.util.Properties)}.
+	 */
+	@Test
+	void testLoadIgnoreCase() {
+		this.envVars.put("CONFIG1_key1", "value11");
+		this.envVars.put("config1_KEY2", "value12");
+		this.envVars.put("config2_kEy1", "value21");
+
+		Properties orig1 = new Properties();
+		Properties orig2 = new Properties();
+
+		orig1.setProperty("key1", "orig11");
+		orig1.setProperty("key2", "orig12");
+		orig2.setProperty("key1", "orig21");
+
+		Properties res1 = this.loader.load("config1", orig1);
+		Properties res2 = this.loader.load("CONFIG2", orig2);
+
+		Assertions.assertEquals("value11", res1.getProperty("key1"));
+		Assertions.assertEquals("value12", res1.getProperty("key2"));
+		Assertions.assertEquals("value21", res2.getProperty("key1"));
+	}
+
+	/**
+	 * Tests that multiple keys can get the same value from the same var
+	 * <p>
+	 * Test method for
+	 * {@link de.unistuttgart.iaas.amyassist.amy.core.configuration.EnvironmentConfigurationLoaderImpl#load(java.lang.String, java.util.Properties)}.
+	 */
+	@Test
+	void testLoadColidingName() {
+		this.envVars.put("a_b_c", "value11");
+
+		Properties orig1 = new Properties();
+		Properties orig2 = new Properties();
+
+		orig1.setProperty("c", "orig1");
+		orig2.setProperty("b_c", "orig2");
+
+		Properties res1 = this.loader.load("a_b", orig1);
+		Properties res2 = this.loader.load("a", orig2);
+
+		Assertions.assertEquals("value11", res1.getProperty("c"));
+		Assertions.assertEquals("value11", res2.getProperty("b_c"));
+	}
+
+	/**
+	 * Tests that load accepts all valid names.
+	 * 
+	 * @param var
+	 *            a name
+	 */
+	@ParameterizedTest
+	@MethodSource("validNames")
+	void testValidNames(String var) {
+		try {
+			this.loader.load(var, new Properties());
+		} catch (IllegalArgumentException e) {
+			Assertions.fail("There should not be an exception. Seed: " + seed);
+		}
+	}
+
+	/**
+	 * Tests that load rejects all invalid names.
+	 * 
+	 * @param var
+	 *            a name
+	 */
+	@ParameterizedTest
+	@MethodSource("invalidNames")
+	void testInvalidNames(String var) {
+		Assertions.assertThrows(IllegalArgumentException.class, () -> this.loader.load(var, new Properties()),
+				"No Exception. Seed: " + seed);
```

Comment:
this is not needed

Reply:
To be able to easier find the String that failed, to be able to debug to issue.

Reply:
the string is logged in the test run because it ist the Parameter of the Parameterized test